server/filestore.go

```diff
@@ -7399,6 +7399,22 @@ func (mb *msgBlock) msgFromBufEx(buf []byte, sm *StoreMsg, hh hash.Hash64, doCop
 	return sm, nil
 }
 
+// SubjectForSeq will return what the subject is for this sequence if found.
+func (fs *fileStore) SubjectForSeq(seq uint64) (string, error) {
+	fs.mu.Lock()
```

Comment:
I think this could be a read lock for fs. And we can release after we get the block.

Reply:
Can it be a read lock if we have to page the block into the cache as a part of this operation?

Reply:
The mb will need write lock possible but we do not need it AFAIK in this function from the fs lock.